src/deepsparse/utils/onnx.py

```diff
@@ -60,15 +60,16 @@ def save_onnx_to_temp_files(model: onnx.ModelProto, with_external_data=False) ->
     :param model: The onnx model to save to temporary directory
     :param with_external_data: Whether to save external data to a separate file
     """
-    shaped_model = tempfile.NamedTemporaryFile(mode="w", delete=False)
-    _LOGGER.info(f"Saving model to temporary directory: {tempfile.tempdir}")
+
+    shaped_model = tempfile.NamedTemporaryFile(suffix=".onnx", delete=False, mode="w")
+    _LOGGER.info(f"Saving model to temporary file: {shaped_model.name}")
 
     if with_external_data:
-        external_data = os.path.join(
-            tempfile.tempdir, next(tempfile._get_candidate_names())
+        external_data = tempfile.NamedTemporaryFile(
+            suffix=".data", delete=False, mode="w"
         )
-        has_external_data = save_onnx(model, shaped_model.name, external_data)
-        _LOGGER.info(f"Saving external data to temporary directory: {external_data}")
+        _LOGGER.info(f"Saving external data to temporary file: {external_data.name}")
+        has_external_data = save_onnx(model, shaped_model.name, external_data.name)
     else:
         has_external_data = save_onnx(model, shaped_model.name)
     try:
```

Comment:
Will the next line ever raise an error? if not I presume the try .. finally block can be removed; specially cause yield is used

Reply:
This was not originally written by me, but the way I read it is:
- yield `shaped_model.name` and keep it "alive" as long as the context manager is open
- once the context manager closes, run everything in the `finally` statement (i.e do cleanup)